bundles/org.openhab.binding.twitter/src/main/java/org/openhab/binding/twitter/internal/TwitterHandler.java

```diff
@@ -0,0 +1,298 @@
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import twitter4j.DirectMessage;
+import twitter4j.ResponseList;
+import twitter4j.Status;
+import twitter4j.StatusUpdate;
+import twitter4j.Twitter;
+import twitter4j.TwitterException;
+import twitter4j.TwitterFactory;
+import twitter4j.auth.AccessToken;
+
+/**
+ * The {@link TwitterHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Scott Hanson - Initial contribution
+ */
+
+@NonNullByDefault
+public class TwitterHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(TwitterHandler.class);
+
+    private @Nullable TwitterConfig config;
+    @SuppressWarnings("unused")
+    private @Nullable ScheduledFuture<?> refreshTask;
+
+    private static final int CHARACTER_LIMIT = 280;
+
+    private static @Nullable Twitter client = null;
+    boolean isProperlyConfigured = false;
+
+    public TwitterHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    // creates list of available Actions
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singletonList(TwitterActions.class);
+    }
+
+    @SuppressWarnings({ "null", "unused" })
+    @Override
+    public void initialize() {
+        config = getConfigAs(TwitterConfig.class);
+
+        try {
+            // create a New Twitter Client
+            client = createClient();
+            // verify client is valid
+            if (client != null) {
+                refresh();// Get latest status
+                isProperlyConfigured = true;
+                refreshTask = scheduler.scheduleWithFixedDelay(this::refresh, 0, config.refresh, TimeUnit.MINUTES);
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE);
+            }
+        } catch (UnhandledException e) {
+            updateStatus(ThingStatus.OFFLINE);
+        }
+    }
+
+    /**
+     * Internal method for Getting Twitter Status
+     *
+     */
+    @SuppressWarnings("null")
+    private void refresh() {
+        try {
+            if (!checkPrerequisites()) {
+                return;
+            }
+
+            ResponseList<Status> statuses = client.getUserTimeline();
+            if (statuses.size() > 0) {
+                updateState(CHANNEL_LASTTWEET, StringType.valueOf(statuses.get(0).getText()));
+            } else {
+                logger.debug("No Statuses Found");
+            }
+        } catch (TwitterException e) {
+            logger.info("Error when trying to refresh Twitter Account: {}", e.getMessage());
```

Comment:
Logging to info should be used sparsely e.g. a newly started component or a user file that has been loaded. This could be debug.